File: image-api/src/model/role.rs
use serde::{ Serialize, Deserialize };

use crate::model::user_permissions::UserPermissions;

#[derive(Default, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Role {
    pub id: u8,
    pub role_name: String,
    pub permissions: UserPermissions,
}

File: dam-fe/src/models/Role.ts
import UserPermissions, { UserPermissionsImpl } from './UserPermissions';

export interface Role {
    id: number,
    roleName: string,
    permissions: UserPermissions,
}

export class RoleImpl implements Role {
    id: number;
    roleName: string;
    permissions: UserPermissions;

    constructor (
        id: number = 0,
        name: string = '',
        permissions: UserPermissions = new UserPermissionsImpl()
    ) {
        this.id = id;
        this.roleName = name;
        this.permissions = permissions;
    }
}

export default Role;

